### PR TITLE
WIP: Go workspaces except gengo changes

### PR DESCRIPTION
**Pull request by thockin**
I am trying to see if there's a useful subset of the 90-commits-and-counting PR to implement Go workspaces that could merge without horking the universe.

/kind cleanup

```release-note
TODO
```

### CODE REVIEW

File: hack/make-rules/clean.sh [hunk 1]
```diff
@@ -24,15 +24,11 @@ source "${KUBE_ROOT}/hack/lib/util.sh"
 CLEAN_PATTERNS=(
   "_tmp"
   "doc_tmp"
-  "((?!staging\/src\/k8s\.io\/apiextensions-apiserver\/pkg\/generated\/openapi).)*/zz_generated.openapi.go"
```
Comment (aojea) [inline hack/make-rules/clean.sh:27]: but  ... this things are no longer cleaned ??

File: hack/update-go-workspace.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+$X
+__EOF__
+    go mod download # generate go.work.sum
+)
+
+# Generate the top-level workspace.
+go work init
+git ls-files -z ':(glob)**/go.mod' \
+    | xargs -0 -n1 dirname \
+    | grep -v "/_" ` # skip directories named _foo ` \
```
Comment (aojea) [inline hack/update-go-workspace.sh:63]: ?

Comment (thockin) [reply hack/update-go-workspace.sh:63]: staging/src/k8s.io/kms/internal/plugins/_mock/go.mod

File: hack/make-rules/clean.sh (hunk 1 above)
Comment (thockin) [reply hack/make-rules/clean.sh:27]: This regex matches nothing and all openapi files are properly commeiited now.

**Comment by thockin** [comment]
I'm not sure this PR will be possible.  Workspaces wants the symlinks gone from `./vendor` but all of the legacy tools need them.

File: hack/update-vendor.sh [hunk 3]
```diff
@@ -86,12 +88,13 @@ function ensure_require_replace_directives_for_all_dependencies() {
 
   # Propagate root replace/require directives into staging modules, in case we are downgrading, so they don't bump the root required version back up
   for repo in $(kube::util::list_staging_repos); do
-    pushd "staging/src/k8s.io/${repo}" >/dev/null 2>&1
+    (
```
[the remark this answers is not in the document]
Comment (thockin) [reply hack/update-vendor.sh:91]: right!  Just easier to read. https://github.com/kubernetes/kubernetes/pull/122230

**Comment by thockin** [comment]
Ugh, I also forgot that `go work vendor` isn't released yet.

**Comment by thockin** [comment]
Closing for now.  There's no way to get both workspaces AND symlinks in vendor to work.  Running tests is a waste of time for the moment.

**Pull request closed without merging**